Scenarios/Authentication/Mixed-auth-allow-anonymous-workaround/Idp/Program.cs

```diff
@@ -0,0 +1,74 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Security.Claims;
+using Microsoft.AspNetCore;
+using Microsoft.AspNetCore.Authentication;
+using Microsoft.IdentityModel.Tokens;
+using OpenIddict.Abstractions;
+using OpenIddict.Server;
+using OpenIddict.Server.AspNetCore;
+
+var builder = WebApplication.CreateBuilder(args);
+
+builder.Services.AddOpenIddict()
+    .AddServer(options =>
+    {
+        options.AddDevelopmentEncryptionCertificate();
+        options.AddDevelopmentSigningCertificate();
+        options.AllowClientCredentialsFlow();
+        options.SetTokenEndpointUris("/token");
+        options.EnableDegradedMode();
+        options.UseAspNetCore();
+        options.RegisterScopes("api");
+        options.DisableAccessTokenEncryption();
+        options.AddEventHandler<OpenIddictServerEvents.ValidateTokenRequestContext>(options =>
+            options.UseInlineHandler(async context =>
+            {
+                if (!context.Request.IsClientCredentialsGrantType())
+                {
+                    context.Reject(error: OpenIddictConstants.Errors.InvalidGrant);
+                    return;
+                }
+
+                if (!string.Equals(context.ClientId, "console_app", StringComparison.Ordinal))
+                {
+                    context.Reject(error: OpenIddictConstants.Errors.InvalidClient);
+                    return;
+                }
+
+                if (!string.Equals(context.ClientSecret, "secret", StringComparison.Ordinal))
```

Comment:
You'll probably want to add a warning indicating this wouldn't be okay in a real world application, where you'd need to use a time-constant comparison to help mitigate statistical attacks (and would probably want some form of hashing or key derivation).

Reply:
done thanks